src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/EventListener.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.whereabouts.services
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.jms.annotation.JmsListener
+import org.springframework.stereotype.Service
+
+@Service
+open class EventListener(@Qualifier("attendanceServiceAppScope") private val attendanceService: AttendanceService) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+    val gson: Gson = GsonBuilder().create()
```

Comment:
should be able to just inject gson instead

Reply:
Yep, no worries. It's copied from offender case notes.

Reply:
Although, not sure we need to inject it, thinking about it. Unless we have a specific reason we want the ability to swap out implementations?

Reply:
well it saves having to build another `Gson` - there is one available already so worth just using it - see https://github.com/ministryofjustice/keyworker-api/blob/master/src/main/java/uk/gov/justice/digital/hmpps/keyworker/events/EventListener.kt